Content.Client/GameObjects/Components/Clothing/ClothingComponent.cs

```diff
@@ -56,6 +57,54 @@ public override void Initialize()
         {
             base.Initialize();
             ClothingEquippedPrefix = ClothingEquippedPrefix;
+
+            // Get parent RSIs to fill in missing sprites
+            if (!Owner.TryGetComponent(out SharedItemComponent? item))
+            {
+                return;
+            }
+
+            var rsiPath = item.RsiPath;
+            if (rsiPath == null)
+            {
+                return;
+            }
+
+            var rsi = IoCManager.Resolve<IResourceCache>().GetResource<RSIResource>(SharedSpriteComponent.TextureRoot / rsiPath).RSI;
+
+            var prototypeManager = IoCManager.Resolve<IPrototypeManager>();
+
+            FillInParentSprites(prototypeManager, Owner.Prototype?.Parent, rsi);
+        }
+
+        private void FillInParentSprites(IPrototypeManager prototypeManager, string? parentPrototypeName, RSI rsi)
+        {
+            if (parentPrototypeName == null)
+            {
+                return;
+            }
+
+            var prototype = prototypeManager.Index<EntityPrototype>(parentPrototypeName);
+            if (!prototype.TryGetComponent("Clothing", out ClothingComponent? parentClothes) || parentClothes.RsiPath == null)
+            {
+                return;
+            }
+
+            var parentRsi = IoCManager.Resolve<IResourceCache>().GetResource<RSIResource>(SharedSpriteComponent.TextureRoot / parentClothes.RsiPath).RSI;
+            var enumerator = parentRsi.GetEnumerator();
+
+            while (enumerator.MoveNext())
```

Comment:
Zumorica I tried putting this in a foreach but... it didn't work. The IDE doesnt seem to think that this enumerator is correct or something